File: src/main/java/net/hectus/neobb/turn/default_game/mob/TPiglin.java
package net.hectus.neobb.turn.default_game.mob;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.HotClazz;
import net.hectus.neobb.turn.default_game.attributes.function.EventFunction;
import org.bukkit.entity.Piglin;
import org.bukkit.potion.PotionEffectType;

import java.util.List;

public class TPiglin extends MobTurn<Piglin> implements EventFunction, HotClazz {
    public TPiglin(NeoPlayer player) { super(player); }
    public TPiglin(Piglin data, NeoPlayer player) { super(data, player); }

    @Override
    public boolean requiresUsageGuide() { return true; }

    @Override
    public int cost() {
        return 3;
    }

    @Override
    public void triggerEvent() {
        data.addScoreboardTag(player.game.id);
    }

    public List<Buff> buffs() {
        return List.of(new Buff.Effect(Buff.BuffTarget.YOU, PotionEffectType.BLINDNESS), new Buff.Effect(Buff.BuffTarget.YOU, PotionEffectType.SLOWNESS), new Buff.Luck(Buff.BuffTarget.YOU, 30));
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/mob/TPiglin.kt
package net.hectus.neobb.modes.turn.default_game.mob

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.Effect
import net.hectus.neobb.buff.Luck
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.HotClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.EventFunction
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.entity.Piglin
import org.bukkit.potion.PotionEffectType

class TPiglin(data: Piglin?, cord: Cord?, player: NeoPlayer?) : MobTurn<Piglin>(data, cord, player), EventFunction, HotClazz {
    override val cost: Int = 3

    override fun triggerEvent() {
        data!!.addScoreboardTag(player!!.game.id)
    }

    fun buffs(): List<Buff<*>> {
        return listOf(Effect(PotionEffectType.BLINDNESS), Effect(PotionEffectType.SLOWNESS), Luck(30))
    }
}
